core/unit_test/CMakeLists.txt

```diff
@@ -659,6 +659,7 @@ if(Kokkos_ENABLE_CUDA)
       UnitTestMainInit.cpp
       ${Cuda_SOURCES1}
       cuda/TestCuda_ReducerViewSizeLimit.cpp
+      cuda/TestCuda_TeamSizeMaxIsPositive.cpp
```

Comment:
Why are we adding a CUDA-specific test and not running it for all backends?

Reply:
Maybe we can but it is probably out of the initial scope of the task. The issue https://github.com/kokkos/kokkos/issues/3371 was about CUDA problem.
@fnrizzi what do you think ?